### PR TITLE
Remove html from csv

### PR DESCRIPTION
**Pull request by jasalisbury**
## Description of change

Convert HTML to plain text in CSV download

## How to test

Download CSVs with complex `context` and `ttaProvided` fields. Verify no HTML is being sent back in the CSV.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-477

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by thewatermethod — COMMENTED** [review]
This successfully strips out HTML tags. 

However, it preserves encoding so special characters like ’ and ñ remain, which I think is what creates the cruft in the Jira ticket.

I'd imagine that if you created something on MS Word and pasted it into a WYSIWYG, it'd still look weird? That's the only thing I can think of to explain the text in the Jira. This wouldn't fix that issue, but maybe thats ok.

**Comment by GarrettEHill** [comment]
> This successfully strips out HTML tags.
> 
> However, it preserves encoding so special characters like ’ and ñ remain, which I think is what creates the cruft in the Jira ticket.
> 
> I'd imagine that if you created something on MS Word and pasted it into a WYSIWYG, it'd still look weird? That's the only thing I can think of to explain the text in the Jira. This wouldn't fix that issue, but maybe thats ok.

The mangled encoding has been moved to a different JIRA.  https://ocio-jira.acf.hhs.gov/browse/TTAHUB-521

File: src/lib/transform.js [hunk 1]
```diff
@@ -62,6 +63,20 @@ function transformRelatedModel(field, prop) {
   return transformer;
 }
 
+function transformHTML(field) {
+  async function transformer(instance) {
+    const html = instance[field] || '';
+    const value = convert(html);
```
Comment (GarrettEHill) [inline src/lib/transform.js:69]: I thin the Regex method would be faster, and allow for the implementation of https://ocio-jira.acf.hhs.gov/browse/TTAHUB-520

Comment (GarrettEHill) [reply src/lib/transform.js:69]: After reading though the libraries examples, it does in fact retain formatting.  Good find.

File: src/lib/transform.test.js [hunk 2]
```diff
@@ -197,6 +197,17 @@ describe('activityReportToCsvRecord', () => {
     expect(output).toMatchObject(expectedOutput);
   });
 
+  it('removes HTML tags from context', async () => {
+    const report = ActivityReport.build({
+      context: '<p data-attribute="test">test text</p>',
```
Comment (GarrettEHill) [inline src/lib/transform.test.js:202]: maybe a few more test might be a good idea, order list, nested  ordered list, unordered list, and table.

**Comment by thewatermethod** [comment]
> The mangled encoding has been moved to a different JIRA. https://ocio-jira.acf.hhs.gov/browse/TTAHUB-521

Whoops, missed that

**Review by thewatermethod — APPROVED** [review]

File: src/lib/transform.js (hunk 1 above)
Comment (GarrettEHill) [reply src/lib/transform.js:69]: Please add table formatting:
`const value = convert(html, {
  tables: true
});`

File: src/lib/transform.test.js [hunk 3]
```diff
@@ -197,6 +197,106 @@ describe('activityReportToCsvRecord', () => {
...
+              <td>Three</td>
+              <td>Four</td>
+            </tr>
+          </tbody>
+        </table>
+        `,
+      });
+      const output = await activityReportToCsvRecord(report);
+      const expectedOutput = {
+        context: 'First Second One Two Three Four',
```
Comment (GarrettEHill) [inline src/lib/transform.test.js:294]: Update expected output to have a formatted table

File: src/lib/transform.js [hunk 4]
```diff
@@ -62,6 +63,20 @@ function transformRelatedModel(field, prop) {
   return transformer;
 }
 
+function transformHTML(field) {
+  async function transformer(instance) {
+    const html = instance[field] || '';
+    const value = convert(html, { tables: true });
```
Comment (thewatermethod) [inline src/lib/transform.js:69]: looks like ```{ tables: true }``` is deprecated, they recommend 
```selectors: [ { selector: 'table', format: 'dataTable' } ]``` instead

https://github.com/html-to-text/node-html-to-text#options

**Review by GarrettEHill — APPROVED** [review]

**Pull request merged**